File: app/src/main/java/com/example/features/AbstractFragment.java
package com.example.features;

import android.support.v4.app.Fragment;

import com.example.App;
import com.example.BuildConfig;
import com.example.toolkit.RetryCallback;
import com.squareup.leakcanary.RefWatcher;

public class AbstractFragment extends Fragment implements PresenterView {
  @Override
  public void onDestroy() {
    if (BuildConfig.DEBUG) {
      RefWatcher refWatcher = App.getRefWatcher();
      refWatcher.watch(this);
    }

    super.onDestroy();
  }

  @Override
  public void showLoading() {

  }

  @Override
  public void loaded() {

  }

  @Override
  public void error(Throwable throwable) {

  }

  @Override
  public void errorWithRetry(Throwable throwable, RetryCallback retryAction) {

  }
}

File: libs/toolkit-android/src/main/kotlin/br/com/vitorsalgado/example/utils/base/AbstractFragment.kt
// package br.com.vitorsalgado.example.android.utils.base
//
// import android.support.v4.app.Fragment
// import br.com.vitorsalgado.example.App
// import br.com.vitorsalgado.example.BuildConfig
// import br.com.vitorsalgado.example.toolkit.RetryCallback
//
// open class AbstractFragment : Fragment(), PresenterView {
//  override fun onDestroy() {
//    if (BuildConfig.DEBUG) {
//      val refWatcher = App.refWatcher
//      refWatcher.watch(this)
//    }
//
//    super.onDestroy()
//  }
//
//  override fun showLoading() {
//
//  }
//
//  override fun loaded() {
//
//  }
//
//  override fun error(throwable: Throwable) {
//
//  }
//
//  override fun errorWithRetry(throwable: Throwable, retryAction: RetryCallback) {
//
//  }
// }
